test/wasi/args.wast

```diff
@@ -0,0 +1,130 @@
+(module
+  (import "wasi_snapshot_preview1" "args_get" (func $wasi_args_get (param i32 i32) (result i32)))
+  (import "wasi_snapshot_preview1" "args_sizes_get" (func $wasi_args_sizes_get (param i32 i32) (result i32)))
+  (memory 1)
+
+  (export "memory" (memory 0))
+  (data (i32.const 32) "Hello" ) ;; Memory[32] = "Hello,"
+  (data (i32.const 48) "World!" ) ;; Memory[48] = "World!"
+
+  (func (export "check_args")(result i32)
+  (local $i i32)
+  (local $mismatched_char_count i32)
+  (local $argc i32)
+  (local $argv i32)
+  (local $argv_buf i32)
+  (local $value_addr i32)
+  (local $expected_addr i32)
+    ;; Memory[0] = args count
+    ;; Memory[4] = args size in characters
+    i32.const 0
+    local.tee $argc
+    i32.const 4
+    call $wasi_args_sizes_get
+    drop
+
+    i32.const 128 ;; argv[] (list of pointers to the strings)
+    local.tee $argv
+    i32.const 192 ;; argv_buf (the buffer the the strings themselves)
+    local.tee $argv_buf
+    call $wasi_args_get
+    drop
+
+    i32.const 0
+    local.set $mismatched_char_count
+
+    ;; Check if each arg matches the expect values of "Hello," and "World!"
+    i32.const 1  ;; start at 1, skip argv[0] (name of currently running file)
```

Comment:
Since the function returns with two buffers in an expected layout, we could check them in one step (without a loop for arguments).